java/arcs/core/allocator/Allocator.kt

```diff
@@ -240,5 +244,47 @@ class Allocator(
                 coroutineContext
             )
         }
+
+        /**
+         * Creates an [Allocator] which does not attempt to serialize Arc/Particle state to storage.
+         *
+         * This is primarily useful for tests, but also may be of limited use in production if Arc
+         * serialization and resurrection is not a requirement for your environment.
+         */
+        @ExperimentalCoroutinesApi
+        fun createNonSerializing(
```

Comment:
If we intend to use this in production we should add a unit test for it. Otherwise we could restructure to make this an interface and allow using a non-serializing test-only one vs the normal one, to allow test only code to be separated and tagged `test_only`. @cromwellian FYI